pkg/kubectl/resource/builder.go

```diff
@@ -567,7 +567,9 @@ func splitResourceTypeName(s string) (resourceTuple, bool, error) {
 // compatible types into individual entries and give them their own items. The original object
 // is not passed to any visitors.
 func (b *Builder) Flatten() *Builder {
-	b.flatten = true
+	if len(b.resources) != 1 {
```

Comment:
making this behave differently depending on the state of `Builder` isn't good

Reply:
@liggitt thanks! Aggree, should not change the Flatten() behavior. Will move the check to Get Run cmd func.